frontend/packages/topology/src/components/list-view/TopologyListViewKindGroup.tsx

```diff
@@ -32,21 +34,29 @@ const TopologyListViewKindGroup: React.FC<TopologyListViewKindGroupProps> = ({
   childNodes.sort((a, b) => a.getLabel().localeCompare(b.getLabel()));
 
   return (
-    <DataListItem key={kind} aria-labelledby={`${kind}_label`} isExpanded>
+    <DataListItem
+      key={kind}
+      aria-labelledby={`${groupLabel}-${resourceLabel.replace(' ', '-')}-label`}
+      isExpanded
+    >
       <DataListItemRow className="odc-topology-list-view__kind-row">
         <DataListItemCells
           dataListCells={[
             <DataListCell
               key={kind}
               className="odc-topology-list-view__kind-label"
-              id={`${kind}_label`}
+              id={`${groupLabel}-${resourceLabel.replace(' ', '-')}-label`}
```

Comment:
Are you sure that resource label can contains spaces, but the group label not? Whats about this small improvement:

```suggestion
              id={`${groupLabel}-${resourceLabel}-label`.replace(' ', '-')}
```

Sure you need to update it above also.